### ISSUE TITLE
False positive "consider-using-dict-items" when iteration over dict.keys()

### ISSUE THREAD

**Issue by Flamefire** [pylint-dev/pylint#9554]
### Bug description

```python
"""TEST module"""
import os

for var in os.environ.keys():
    if var.startswith('foo_'):
        del os.environ[var]
```

The loop triggers both "consider-using-dict-items" & "consider-iterating-dictionary". The former is a false positive.

This is especially annoying when "consider-iterating-dictionary" is disabled because someone insists that `.keys()` is clearer

However fixing the 2nd warning by removing `.keys()` both go away

### Command used

```shell
pylint test.py 
pylint --disable=C0201 test.py
```

### Pylint output

```python
************* Module test
test.py:4:0: C0206: Consider iterating with .items() (consider-using-dict-items)
test.py:4:11: C0201: Consider iterating the dictionary directly instead of calling .keys() (consider-iterating-dictionary)

------------------------------------------------------------------
Your code has been rated at 7.14/10 (previous run: 5.71/10, +1.43)
```

### Expected behavior

No "consider-using-dict-items" warning

### Pylint version

```shell
pylint 3.1.0
astroid 3.1.0
Python 3.10.12 (main, Nov 20 2023, 15:14:05) [GCC 11.4.0]
```
